editor/editor_properties.cpp

```diff
@@ -3602,6 +3603,31 @@ EditorProperty *EditorInspectorDefaultPlugin::get_editor_for_property(Object *p_
 				Vector<String> options = p_hint_text.split(",", false);
 				editor->setup(options, false, (p_hint == PROPERTY_HINT_ENUM_SUGGESTION));
 				return editor;
+			} else if (p_hint == PROPERTY_HINT_INPUT_NAME) {
+				EditorPropertyTextEnum *editor = memnew(EditorPropertyTextEnum);//should probably use a better editor gui with a searchbar
```

Comment:
Given what you've expressed on [RocketChat](https://chat.godotengine.org/channel/input?msg=NsbTKSbSh7oFGnenX), this should at least be a TODO:
```suggestion
				// TODO: Should probably use a better editor GUI with a search bar.
				EditorPropertyTextEnum *editor = memnew(EditorPropertyTextEnum);
```